src/action.js

```diff
@@ -0,0 +1,189 @@
+import * as os from "os";
+
+import * as github from "@actions/github";
+import * as toolCache from "@actions/tool-cache";
+import * as core from "@actions/core";
+import * as exec from "@actions/exec";
+
+// Define consts
+const GRUNTWORK_GITHUB_ORG = "gruntwork-io";
+const PATCHER_GITHUB_REPO = "patcher-cli";
+const PATCHER_VERSION = "v0.4.3";
+
+const PATCHER_BINARY_PATH = "/tmp/patcher"
+
+const REPORT_COMMAND = "report";
+const UPDATE_COMMAND = "update";
+
+const NON_INTERACTIVE_FLAG = "--non-interactive"
+const NO_COLOR_FLAG = "--no-color"
+const SKIP_CONTAINER_FLAG = "--skip-container-runtime"
+
+function osPlatform() {
+    switch (os.platform()) {
+        case "linux":
+            return "linux";
+        case "darwin":
+            return "darwin";
+        default:
+            core.setFailed("Unsupported operating system - the Patcher action is only released for Darwin and Linux");
+            return;
+    }
+}
+
+async function openPullRequest(patcherRawOutput, dependency, ghToken) {
+    const head = `patcher-updates-${dependency}`
+    const title = `[Patcher] Update ${dependency}`
+    const commitMessage = "Update dependencies using Patcher"
+    const commitAuthor = "Grunty"
+    const commitEmail = "grunty@gruntwork.io"
+
+    const body = `
+Updated the \`${dependency}\` dependency using Patcher.
+
+### Update summary
+\`\`\`yaml
+${patcherRawOutput}
+\`\`\`
+`
+
+    await exec.exec("git", ["config", "user.name", commitAuthor])
+    await exec.exec("git", ["config", "user.email", commitEmail])
+    await exec.exec("git", ["add", "."])
+    await exec.exec("git", ["commit", "-m", commitMessage])
+    await exec.exec("git", ["checkout", "-b", head])
+
+    const context = github.context;
+
+    await exec.exec("git", ["push", "-f", `https://${ghToken}@github.com/${context.repo.owner}/${context.repo.repo}.git`])
```

Comment:
What if branch already exists in remote?

Reply:
It will force push. I will create a follow up task to fix this. It should pull the branch before doing any changes.
e.g:
1. if the user is in the middle of doing the updates
1. they already deleted the README-TO-COMPLETE.md (updated to v0.1.0)
1. workflows runs the action again 
1. there is a new minor version for the module (v0.1.1)
1. The action just continue the work from there